### DISCUSSION TITLE
Select all / deselect all toggle - multiselect dropdown

### DISCUSSION THREAD

**Discussion by RedOchrePM** [anvilistas/anvil-extras#241] [Ideas]
It would be useful to be able to toggle all-on / all-off, especially for long dropdown lists:

![2021-10-21_18-59-56](https://user-images.githubusercontent.com/71299741/138265073-603d958c-9dc7-4011-a17e-9c88d787a206.png)

**Comment by s-cork** [comment]
Do you fancy writing the pr?
It's a fairly easy implementation.
(Approximately 4 lines of code)

https://developer.snapappointments.com/bootstrap-select/methods/

You just have to hook into the api above with a `select_all` and `deselect_all` method in anvil.
You can see this working with the unsupported api.
```python
self.multil_selecet_drop_down._el.selectpicker("selectAll")
```

**Comment by RedOchrePM** [comment]
Um, I’m not a developer… happy to have a go, but don’t want to break anything.

I’ve also just realised that the .selectpicker('render') method would be super useful.

**Comment by s-cork** [comment]
@RedOchrePM - No pressure - it just seems like a nice first pr if that's not something you're used to.

You'll want to follow the steps in the [Contribution guide](https://anvil-extras.readthedocs.io/en/latest/guides/contributing.html).
If you get stuck just add to this thread.

**Comment by theref** [comment]
Hi, I thought I'd have a go at getting this added. From looking at the current code, I believe I need something along the lines of:
```
    def select_all(self, **event_args):
        self._el.selectpicker("selectAll")

    def deselect_all(self, **event_args):
        self._el.selectpicker("deselectAll")
```
added to the `MultiSelectDropDown` class. I'm not sure how to add in buttons within the dropdown for this functionality though.

**Reply by s-cork** [reply]
If you look at the examples from here - https://developer.snapappointments.com/bootstrap-select/examples/

There's an example that has those buttons using an attribute we can add to the initialisation of the component. You might see some other examples in the set up for the component.

Not sure what the property should be called.  `include_select_buttons` seems a bit long.

**Reply by theref** [reply]
Got it working and have submitted a PR. Thanks for your help!
